### PR TITLE
Add polygon primitive

### PR DESCRIPTION
**Pull request by mdales**
I was looking to use this for a specific application that needs polygon support, so I've added a basic Polygon shape type. 

I manually tested this under both macOS 13 and Fedora 37, and the example in the snippet works (tested both filled and unfilled).

### CODE REVIEW

File: Sources/CairoGraphics/context/CairoContext.swift [hunk 1]
```diff
@@ -100,6 +100,26 @@ public final class CairoContext: GraphicsContext {
...
+        guard polygon.points.count > 0 else {
+            return
+        }
+
+        markImageAsUnflushed()
+        context.setSource(color: polygon.color.asDoubleTuple)
+
+        context.move(to: polygon.points.first!.asTuple)
+        for point in polygon.points.suffix(polygon.points.count - 1) {
+            context.line(to: point.asTuple)
```
Comment (fwcd) [inline Sources/CairoGraphics/context/CairoContext.swift]: I think, we should close off the path, at least when filling is disabled (or, if this is intended, perhaps rename `Polygon` to something more general, e.g. `Path`). Same goes for the CoreGraphics implementation.

**Review by fwcd — COMMENTED** [review]
Neat, thanks for looking into this. Looking good for the most part, just one remark below

File: Sources/CairoGraphics/context/CairoContext.swift (hunk 1 above)
Comment (mdales) [reply Sources/CairoGraphics/context/CairoContext.swift]: Good idea. I did the change in the Polygon itself, so the logic was just in one place, and added some tests. I also added a check that it wasn't already closed, which was the case for the geospatial data I'm using this to render.

**Comment by mdales** [comment]
Whilst on vacation, I thought more about your comment @fwcd, and realised that an open polygon is not really a polygon, and so now close them regardless of whether they're filled or not. If people want a path, then a new shape could be added for that.

File: Sources/Graphics/shape/Polygon.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import Utils
+
+public struct Polygon<T: IntExpressibleAlgebraicField> {
+    public let paths: [[Vec2<T>]]
```
Comment (fwcd) [inline Sources/Graphics/shape/Polygon.swift:4]: Do we want polygons to be able to represent disjoint shapes? Is there any advantage over just creating multiple polygons in that case?

Comment (mdales) [reply Sources/Graphics/shape/Polygon.swift:4]: This is to let you create polygons with holes in (i.e., you have a void in your polygon where the data already in the image shows through). Both CoreGraphics and Cairo let you provide multiple subpaths and when you then call fill, it'll use either the winding rule or the even/odd method to create the holes. 

If you want to overlap or have individual polygons you shouldn't use this method to stack them up, you should just make separate calls to draw(polygon:...) for that.

It's a bit of an odd interface, but both Cairo and CoreGraphics do the same thing here essentially. The only other API for this I've seen in use is where the first path in the list is treated as the outside, and every subsequent path is assumed to be a hole, but that's now how either of the supported backends work, so I didn't go for that.

Comment (fwcd) [reply Sources/Graphics/shape/Polygon.swift:4]: Ah okay, thanks for clarifying, that sounds reasonable!

**Comment by fwcd** [comment]
Could you rebase this on top of `main` (and perhaps the other PR too)? That way I could run CI on your branches

**Comment by mdales** [comment]
Rebased! I'll do the other one tomorrow as part of working in your feedback 👍

**Comment by fwcd** [comment]
Thanks!

**Pull request merged**